adding TV Shows to Restaurants results in broken activeRecipe

Trying to load [demo] TV shows fails and outputs the following in the console
```
synthetic-storage.js:132 Error parsing manifest at synthetic://arc/handles/firebase://arcs-storage.firebaseio.com/AIzaSyBme42moeI-2k8WgXh-6YK_wYyjEXo4Oz8/0_6_0/chenbrian--LVyQm3hfYSNO7wjCA88:
Post-parse processing error caused by 'undefined' line 401.
Hosted particle 'TVMazeShowTile' does not match interface 'hostedParticle'
      hostedParticle = TVMazeShowTile
```

Repro steps:
1) Visit http://localhost:8786/shells/web-shell/index.html as a newly created user.
2) Enter * in the search bar.
3) Try clicking on [demo] TV show from the search list.

--- a/src/runtime/interface-info.ts
+++ b/src/runtime/interface-info.ts
@@ -124,10 +124,8 @@ export class InterfaceInfo {
 
   _handlesToManifestString() {
     return this.handles
-      .map(handle => {
-        const type = handle.type.resolvedType();
-        return `  ${handle.direction ? handle.direction + ' ': ''}${type.toString()} ${handle.name ? handle.name : '*'}`;
-      }).join('\n');
+      .map(h => `  ${h.direction ? h.direction + ' ': ''}${h.type.toString()} ${h.name ? h.name : '*'}`)
+      .join('\n');
   }
 
   _slotsToManifestString() {
@@ -140,8 +138,7 @@ export class InterfaceInfo {
   toString() {
     return `interface ${this.name}
 ${this._handlesToManifestString()}
-${this._slotsToManifestString()}
-`;
+${this._slotsToManifestString()}`;
   }
 
   static fromLiteral(data) {

--- a/src/runtime/test/type-test.ts
+++ b/src/runtime/test/type-test.ts
@@ -292,6 +292,30 @@ describe('types', () => {
     });
   });
 
+  describe('serialization', () => {
+    it('serializes interfaces', () => {
+      const entity = EntityType.make(['Foo'], {value: 'Text'});
+      const variable = TypeVariable.make('a', null, null);
+      const iface = InterfaceType.make('i', [{type: entity, name: 'foo'}, {type: variable}], [{name: 'x', direction: 'consume'}]);
+      assert.equal(iface.interfaceInfo.toString(),
+`interface i
+  Foo {Text value} foo
+  ~a *
+  consume x `);
+    });
+
+    // Regression test for https://github.com/PolymerLabs/arcs/issues/2575
+    it('disregards type variable resolutions in interfaces', () => {
+      const variable = TypeVariable.make('a', null, null);
+      variable.variable.resolution = EntityType.make(['Foo'], {value: 'Text'});
+      const iface = InterfaceType.make('i', [{type: variable}], []);
+      assert.equal(iface.interfaceInfo.toString(),
+`interface i
+  ~a *
+`);
+    });
+  });
+
   describe('integration', () => {
     const manifestText = `
       schema Product

